weaviate/logger.py

```diff
@@ -1,5 +1,250 @@
 import os
-from logging import getLogger
+import json
+import logging
+import textwrap
+from typing import Dict, Any, Optional, Union, Callable
 
-logger = getLogger("weaviate-client")
-logger.setLevel(os.getenv("WEAVIATE_LOG_LEVEL", "INFO"))
+import httpx
+import grpc
+from grpc.aio import UnaryUnaryCall, UnaryStreamCall, StreamUnaryCall, StreamStreamCall  # type: ignore
+
+def _setup_logger():
+    """Set up the weaviate-client logger with proper configuration."""
+    logger = logging.getLogger("weaviate-client")
+    log_level = os.getenv("WEAVIATE_LOG_LEVEL", "INFO")
+    logger.setLevel(getattr(logging, log_level))
+    logger.propagate = True  # Ensure log propagation is enabled
+    
+    # Add a StreamHandler if none exists
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        handler.setLevel(getattr(logging, log_level))
+        formatter = logging.Formatter('%(message)s')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+    
+    print(f"DEBUG: Logger initialized with level {logger.level}, handlers: {logger.handlers}")
```

Comment:
This is a `print` statement and not `logger.debug` and so will show up regardless of the `WEAVIATE_LOG_LEVEL` specification making the client noisy